src/terrain/terrain.js

```diff
@@ -260,7 +260,8 @@ export class Terrain extends Elevation {
         this._sourceTilesOverlap = {};
         this.proxySourceCache = new ProxySourceCache(style.map);
         this.orthoMatrix = mat4.create();
-        mat4.ortho(this.orthoMatrix, 0, EXTENT, 0, EXTENT, 0, 1);
+        const epsilon = this.painter.transform.projection.name === 'globe' ?  .015 : 0; // Experimentally the smallest value to avoid rendering artifacts (https://github.com/mapbox/mapbox-gl-js/issues/11975)
```

Comment:
Do we also need to check for terrain here?

Reply:
I'm not seeing the missing tile with terrain (no globe) with this fix, but will test out on various devices! I also hit a [failed render test](https://output.circle-artifacts.com/output/job/ae72b852-170d-4096-b020-ab2f4e1a6bd1/artifacts/0/test/integration/render-tests/index.html) when I removed the globe check (no significant difference though)

Reply:
Here's a [codesandbox](https://codesandbox.io/s/young-feather-bphr2q?file=/index.html) with this change